source/MaterialXTest/ShaderGen.cpp

```diff
@@ -146,6 +146,37 @@ bool requiresImplementation(const mx::NodeDefPtr nodeDef)
     return !typeAttribute.empty() && typeAttribute != TYPE_NONE;
 }
 
+// Identifiers for supported light types.
+enum LightTypes
+{
+    DIRECTIONAL_LIGHT,
+    POINT_LIGHT
+};
+
+// Light shader nodes to bind to each light type.
+const std::vector<std::pair<int, std::string>> LIGHT_SHADERS =
+{
+    { DIRECTIONAL_LIGHT, "ND_directionallight" },
+    { POINT_LIGHT, "ND_pointlight" }
+};
+
+// Bind the supported light types to corresponding light shaders
+// for the given shader generator.
+void bindLightShaders(mx::DocumentPtr document, mx::HwShaderGenerator& shadergen)
```

Comment:
Example helper function to bind light shaders before generating surface shaders.